File: buzzvil-sdk-ios/buzzvil-sdk-ios-objc/BenefitHub/BenefitHubViewController.m
@import BuzzvilSDK;
#import "BenefitHubViewController.h"

@interface BenefitHubViewController ()

@end

@implementation BenefitHubViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

// 베네핏허브 표시하기
- (void)showBenefitHub {
  BZVBenefitHub *benefitHub = [BZVBenefitHub benefitHubWithBlock:^(BZVBenefitHubBuilder * _Nonnull builder) {}];
  [benefitHub showOn:self];
}

// 베네핏허브 표시하기
- (void)showBenefitHubWithUnitID {
  BZVBenefitHub *benefitHub = [BZVBenefitHub benefitHubWithBlock:^(BZVBenefitHubBuilder * _Nonnull builder) {
    builder.config = [BZVBenefitHubConfig configWithBlock:^(BZVBenefitHubConfigBuilder * _Nonnull builder) {
      builder.unitID = @"SECOND_FEED_UNIT_ID";
    }];
  }];
  [benefitHub showOn:self];
}

// 적립 가능한 포인트 표시하기
- (void)getAvailiableReward {
  BZVBenefitHub *benefitHub = [BZVBenefitHub benefitHubWithBlock:^(BZVBenefitHubBuilder * _Nonnull builder) {}];
  [benefitHub loadOnSuccess:^() {
      // 적립 가능한 포인트를 직접 구현한 UI에 업데이트합니다.
    } onFailure:^(NSError * _Nonnull error) {
      // 적립 가능한 포인트를 가져올 수 없는 경우
  }];
}

@end

File: buzzvil-sdk-ios/buzzvil-sdk-ios-swift/BenefitHub/BenefitHubViewController.swift
import UIKit
import BuzzvilSDK

class BenefitHubViewController: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    
    showBenefitHub()
  }
  
  // 베네핏허브 표시하기
  func showBenefitHub() {
    let benefitHub = BZVBenefitHub { _ in }
    
    benefitHub.show(on: self)
  }
  
  // 베네핏허브 표시하기
  func showBenefitHubWithUnitID() {
    let benefitHub = BZVBenefitHub { builder in
      builder.config = BZVBenefitHubConfig { builder in
        builder.unitID = "SECOND_BENEFIT_HUB_UNIT_ID"
      }
    }
    benefitHub.show(on: self)
  }
  
  // 적립 가능한 포인트 표시하기
  func getAvailiableReward() {
    let benefitHub = BZVBenefitHub { _ in }
    benefitHub.load(
      onSuccess: {
        // 광고 로드에 성공했을 경우
        // 적립 가능한 포인트를 직접 구현한 UI에 업데이트합니다.
        let availiableReward = benefitHub.availableRewards
      },
      onFailure: { error in
        // 적립 가능한 포인트를 가져올 수 없는 경우
      }
    )
  }
}
